codalab/worker_manager/worker_manager.py

```diff
@@ -75,7 +76,10 @@ def start_worker_job(self):
 
     def run_loop(self):
         while True:
-            self.run_one_iteration()
+            try:
+                self.run_one_iteration()
+            except Exception:
```

Comment:
What kind of exceptions happen?

I'm a bit worried that if we indiscriminately catch exceptions, we won't fix the root issues, and since this is in a loop, whatever failure get will just continue to happen.

Reply:
If you, say, upgrade the codalab instance, you'll briefly get a 404 or other ConnectionRefused error. Maybe we should just try / except this one snippet (https://github.com/codalab/codalab-worksheets/blob/master/codalab/worker_manager/worker_manager.py?rgh-link-date=2020-06-17T07%3A33%3A42Z#L89-L91), and return if it fails (so it can sleep and just retry?)

Reply:
I see.  Maybe we can pull out the factor out the logic used by the worker to connect and have everyone use that.

Reply:
Oh, it's not exactly the same logic except in the abstract sense—the worker / bundlemanager all just have:

```
while True / not exiting:
    try:
        do_some_action()
    except Exception:
        raise
    sleep()
```

see, e.g., https://github.com/codalab/codalab-worksheets/blob/ce690f5ab735b0fae50aecf2b00c3b2b527967aa/codalab/server/bundle_manager.py#L78-L84

The workermanager isn't doing any check-ins like the worker does. It just needs to fetch how many bundles the user currently has staged. If codalab is momentarily down, this fetch will fail and the entire program will exit. You're right that I should be able to just ignore the connection errors, though—I'll change this PR to do so.